app/models/doi.rb

```diff
@@ -1836,6 +1837,12 @@ def event=(value)
     self.send(value) if %w(register publish hide show).include?(value)
   end
 
+  def check_url
```

Comment:
I like this approach, makes it much easier. You should check whether the the DOI is in draft state, when any DOI should be allowed.